cmd/server.go

```diff
@@ -204,11 +210,15 @@ func loadAndVerifyServerConfig(cmd *cobra.Command) (*config.ServerConfig, error)
 	}
 	for _, ep := range strings.Split(serverPdAddr, ",") {
 		if conf.Security.IsTLSEnabled() {
-			if strings.Index(ep, "http://") == 0 {
+			if strings.HasPrefix(ep, HTTP) {
 				return nil, cerror.ErrInvalidServerOption.GenWithStack("PD endpoint scheme should be https")
 			}
-		} else if strings.Index(ep, "http://") != 0 {
-			return nil, cerror.ErrInvalidServerOption.GenWithStack("PD endpoint scheme should be http")
+		} else {
+			if strings.HasPrefix(ep, HTTPS) {
+				return nil, errors.New("PD endpoint scheme is https, please provide certificate")
+			} else if !strings.HasPrefix(ep, HTTP) {
+				return nil, cerror.ErrInvalidServerOption.GenWithStack("PD endpoint scheme should be http")
+			}
```

Comment:
Seems the same code was duplicated, and one has the error code but the other hasn't.

Reply:
Good catch! Fixed.